crates/test-collector/src/lib.rs

```
@@ -338,59 +338,58 @@ fn extract_fork_config_from_args(db: &dyn SyntaxGroup, attr: &Attribute) -> Opti
+    let block_id_value = try_get_block_id(db, &block_id_type, &expr)?;
 
-    Some(RawForkConfig::Params(RawForkParams { url, block_id }))
+    Some(RawForkConfig::Params(RawForkParams {
+        url,
+        block_id_type,
+        block_id_value,
+    }))
 }
 
-fn try_get_block_id(db: &dyn SyntaxGroup, block_id_type: &str, expr: &Expr) -> Option<BlockId> {
+fn try_get_block_id(db: &dyn SyntaxGroup, block_id_type: &str, expr: &Expr) -> Option<String> {
     match block_id_type {
         "Number" => {
             if let Expr::Literal(value) = expr {
-                return Some(BlockId::Number(
-                    u64::try_from(value.numeric_value(db).unwrap()).ok()?,
-                ));
+                return Some(
+                    u64::try_from(value.numeric_value(db).unwrap())
+                        .ok()?
+                        .to_string(),
+                );
             }
-            None
         }
         "Hash" => {
+            // TODO #1179: add range check
             if let Expr::Literal(value) = expr {
-                return Some(BlockId::Hash(
-                    Felt252::from(value.numeric_value(db).unwrap()).into_(),
-                ));
+                return Some(value.numeric_value(db).unwrap().to_string());
             }
-            None
         }
         "Tag" => {
             if let Expr::Path(block_tag) = expr {
-                let tag = block_tag
-                    .elements(db)
-                    .last()
-                    .unwrap()
-                    .identifier(db)
-                    .to_string();
-                return match tag.as_str() {
-                    "Latest" => Some(BlockId::Tag(BlockTag::Latest)),
-                    _ => None,
-                };
+                let tag_elements = block_tag.elements(db);
+                if tag_path_is_valid(&tag_elements, db) {
+                    return Some("Latest".to_string());
+                }
```

Probably should throw an error otherwise?

---

We do, we exit the match statement resulting in returning `None` at the bottom of the function, which then later appends stuff to diagnostic errors

---

This is very convoluted, error should be returned by the validating function.

---

It has been written this way since the test collector existence, it is basically a copied approach from cairo test

---

This can be changed later, but I'd do that only on the new collector side. No point in upgrading it here.

---

#1305